File: backend/src/main/java/com/wecp/progressive/entity/Teacher.java
package com.wecp.progressive.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "teacher")
public class Teacher implements Comparable<Teacher> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "teacher_id")
    private int teacherId;

    @Column(name = "full_name")
    private String fullName;

    @Column(name = "subject")
    private String subject;

    @Column(name = "contact_number")
    private String contactNumber;

    @Column(name = "email")
    private String email;

    @Column(name = "years_of_experience")
    private int yearsOfExperience;

    public Teacher() {
    }

    public Teacher(int teacherId, String fullName, String subject, String contactNumber, String email,
            int yearsOfExperience) {
        this.teacherId = teacherId;
        this.fullName = fullName;
        this.subject = subject;
        this.contactNumber = contactNumber;
        this.email = email;
        this.yearsOfExperience = yearsOfExperience;
    }

    public int getTeacherId() {
        return teacherId;
    }

    public void setTeacherId(int teacherId) {
        this.teacherId = teacherId;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getContactNumber() {
        return contactNumber;
    }

    public void setContactNumber(String contactNumber) {
        this.contactNumber = contactNumber;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public int getYearsOfExperience() {
        return yearsOfExperience;
    }

    public void setYearsOfExperience(int yearsOfExperience) {
        this.yearsOfExperience = yearsOfExperience;
    }

    @Override
    public int compareTo(Teacher o) {
        return Integer.compare(this.yearsOfExperience, o.getYearsOfExperience());
    }
}

File: frontend/src/app/educonnect/models/Teacher.ts
export class Teacher {
    teacherId: number;
    fullName: string;
    contactNumber: string;
    email: string;
    subject: string;
    yearsOfExperience: number;
  
    constructor(teacherId: number, fullName: string, contactNumber: string, email: string, subject: string, yearsOfExperience: number
    ) {
      this.teacherId = teacherId;
      this.fullName = fullName;
      this.contactNumber = contactNumber;
      this.email = email;
      this.subject = subject;
      this.yearsOfExperience = yearsOfExperience;
    }
  
    logAttributes(): void {
      console.log('teacherId:', this.teacherId);
      console.log('fullName:', this.fullName);
      console.log('subject:', this.subject);
      console.log('contactNumber:', this.contactNumber);
      console.log('email:', this.email);
      console.log('yearsOfExperience:', this.yearsOfExperience);
    }
  }